admin-portal/src/pages/aidPackage/aidPackage.tsx

```diff
@@ -30,28 +37,74 @@ export function CreateAidPackage() {
   const [aidPackages, setAidPackages] = useState<AidPackages>({});
 
   useEffect(() => {
-    fetchMedicalNeeds().then((response) => {
-      setMedicalNeeds(response.medicalNeedInfo);
+    MedicalNeedsService.getMedicalNeeds().then((response: any) => {
+      const needsArray = response.data;
+      setMedicalNeeds(needsArray);
       setNeedAssignments(
-        response.medicalNeedInfo.reduce(
-          (previousValue: NeedAssignments, currentValue) => {
+        needsArray.reduce(
+          (previousValue: NeedAssignments, currentValue: any) => {
             previousValue[currentValue.needID] = new Map();
             return previousValue;
           },
           {}
         )
       );
     });
-
-    AidPackageService.getAidPackages().then((res) => {
-      console.log(JSON.stringify(res));
-    });
   }, []);
 
   const goToStep = (step: STEPS) => {
     setCurrentFormStep(step);
   };
 
+  const handleAidPkgPublish = async (supplierID: number): Promise<any> => {
+    const aidPackage = aidPackages[supplierID];
+    const needs = Object.keys(needAssignments)
+      .filter((needID) => {
+        return needAssignments[needID].has(supplierID);
+      })
+      .map((id) => {
+        return {
+          id: Number(id),
+          quantity: needAssignments[id].get(supplierID),
+        };
+      });
+
+    if (aidPackage) {
+      return AidPackageService.postAidPackage({
+        name: aidPackage.name,
+        description: aidPackage.details,
+        aidPackageItems: needs.map((need) => {
+          const medicalNeed = medicalNeeds.find(
+            (currentNeed) => currentNeed.needID === need.id
+          );
+          const supplierQuotationID = medicalNeed?.supplierQuotes.find(
+            (quote) => quote.supplierID === supplierID
+          )?.quotationID;
+
+          return {
+            needID: need.id,
+            quantity: need.quantity,
+            quotationID: supplierQuotationID as number,
+          };
+        }),
+      })
+        .then(() => {
```

Comment:
Let's use Async/await instead of `then`. 
Just to maintain the consistency in the codebase.